coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java

```diff
@@ -67,6 +67,23 @@ public static CoralSpark create(RelNode irRelNode) {
     return new CoralSpark(baseTables, sparkUDFInfos, sparkSQL);
   }
 
+  /**
+   * Users use this function as the main API for getting CoralSpark instance.
+   * This should be used when user need to explicitly add alias for the output SparkSQL string
+   *
+   * @param irRelNode An IR RelNode for which CoralSpark will be constructed.
+   * @param aliases A list of aliases for the translated SparkSQL select list
+   * @return [[CoralSparkInfo]]
+   */
+  public static CoralSpark createWithAlias(RelNode irRelNode, List<String> aliases) {
```

Comment:
I do not think Coral needs an external component to let it know how the alias is. This is like depending on Spark to get something from Coral-Schema. My suggestion is to preserve the old API and find from the current SparkSession whether we are in Spark 2.3 or Spark 3.1. If in Spark 3.1, return aliases from Coral schema.

Reply:
My thought is that at the coral and spark integration site we are getting schema from coral via `ViewToAvroSchemaConverter`, so we can get the column names from coral schema there easily.

Also, I explicitly designed the API to achieve loose coupling and provide flexibility to the client.

Reply:
I understand that. I am saying it is not ideal. Let us discuss if needed. cc: @funcheetah.

Reply:
Yes we can have a discussion on the pros/cons of each approach respectively.